src/utils/time/index.ts

```diff
@@ -67,3 +76,88 @@ function _addYears(date: Date, years: number) {
     result.setFullYear(date.getFullYear() + years);
     return result;
 }
+
+/*
+generates data component for date in format YYYYMMDD-HHMMSS
+ */
+export function parseDateFromFusedDateString(dateTime: string) {
+    const dateComponent: DateComponent<string> = {
+        year: dateTime.slice(0, 4),
+        month: dateTime.slice(4, 6),
+        day: dateTime.slice(6, 8),
+        hour: dateTime.slice(9, 11),
+        minute: dateTime.slice(11, 13),
+        second: dateTime.slice(13, 15),
+    };
+    return getDateFromComponents(dateComponent);
+}
+
+/* sample date format = 2018-08-19 12:34:45
+ the date has six symbol separated number values
+ which we would extract and use to form the date
+ */
+export function tryToParseDateTime(dateTime: string): Date {
+    let dateComponent: DateComponent<string> = {
+        year: null,
+        month: null,
+        day: null,
+        hour: null,
+        minute: null,
+        second: null,
+    };
+
+    dateComponent = getDateComponentsFromSymbolJoinedString(dateTime);
+    if (isDateComponentValid(dateComponent)) {
+        return getDateFromComponents(dateComponent);
+    } else if (
+        dateComponent.year.length === 8 &&
+        dateComponent.month.length === 6
+    ) {
+        // the filename has size 8 consecutive and then 6 consecutive digits
+        // high possibility that the it is some unhandled date time encoding
+        const possibleDateTime = dateComponent.year + '-' + dateComponent.month;
+        return parseDateFromFusedDateString(possibleDateTime);
+    } else {
+        return null;
+    }
+}
+
+function getDateComponentsFromSymbolJoinedString(
+    dateTime: string
+): DateComponent<string> {
+    const [year, month, day, hour, minute, second] = dateTime.match(/\d+/g);
+
+    return { year, month, day, hour, minute, second };
+}
+
+//  has length number of digits in the components
+function isDateComponentValid(dateComponent: DateComponent<string>) {
+    return (
+        dateComponent.year.length === 4 &&
+        dateComponent.month.length === 2 &&
+        dateComponent.day.length === 2
+    );
+}
+
+function parseDateComponentToNumber(
+    dateComponent: DateComponent<string>
+): DateComponent<number> {
+    return {
+        year: parseInt(dateComponent.year),
+        month: parseInt(dateComponent.month) - 1,
```

Comment:
document reason for -1? Also,  dateComponent.month.length === 2, this check will pass for month like Jan/Feb? 
May be little more documentation for this might help in understanding the code in the future :)

Reply:
yes so everyone follows the format YYYY-MM-DD for the date and since its a string substring "02" will remain "02"
and hence have length of 2